src/compiler/binder.ts

```diff
@@ -1399,7 +1397,7 @@ namespace ts {
 
             const typeLiteralSymbol = createSymbol(SymbolFlags.TypeLiteral, "__type");
             addDeclarationToSymbol(typeLiteralSymbol, node, SymbolFlags.TypeLiteral);
-            typeLiteralSymbol.members = { [symbol.name]: symbol };
+            typeLiteralSymbol.members = singletonMap(symbol.name, symbol);

```

Comment:
Nothing about this is like a `singleton` in the design pattern sense, the name is a bit confusing. `createStringKeyedMap` would be more straightforward, if you must wrap the constructor.